packages/plugin/webpack/src/WebpackPlugin.ts

```diff
@@ -283,25 +284,51 @@ the generated files). Instead, it is ${JSON.stringify(pj.main)}`);
   };
 
   launchRendererDevServers = async (logger: Logger): Promise<void> => {
-    const config = await this.configGenerator.getRendererConfig(this.config.renderer.entryPoints);
-    if (config.length === 0) {
+    const configs = await this.configGenerator.getRendererConfig(this.config.renderer.entryPoints);
+    if (configs.length === 0) {
       return;
     }
 
-    for (const entryConfig of config) {
+    const preloadPlugins: string[] = [];
+    let preloadEntriesWithConfig = 0;
+    for (const entryConfig of configs) {
       if (!entryConfig.plugins) entryConfig.plugins = [];
       entryConfig.plugins.push(new ElectronForgeLoggingPlugin(logger.createTab(`Renderer Target Bundle (${entryConfig.target})`)));
 
+      const filename = entryConfig.output?.filename as string;
+      if (filename?.endsWith('preload.js')) {
+        let name = `entry-point-preload-${entryConfig.target}`;
+        if (preloadPlugins.includes(name)) {
+          name = `${name}-${++preloadEntriesWithConfig}`;
+        }
+        entryConfig.plugins.push(new EntryPointPreloadPlugin({ name }));
+        preloadPlugins.push(name);
+      }
+
       entryConfig.infrastructureLogging = {
         level: 'none',
       };
       entryConfig.stats = 'none';
     }
 
-    const compiler = webpack(config);
+    const compiler = webpack(configs);
+
+    const promises = preloadPlugins.map((preloadPlugin) => {
+      return new Promise((resolve, reject) => {
+        compiler.hooks.done.tap(preloadPlugin, (stats) => {
+          if (stats.hasErrors()) {
+            return reject(new Error(`Compilation errors in the preload: ${stats.toString()}`));
+          }
+          return resolve(undefined);
+        });
+      });
+    });
+
     const webpackDevServer = new WebpackDevServer(this.devServerOptions(), compiler);
     await webpackDevServer.start();
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Can we double check that this isn't a valid error w/ null? If it isn't, I'm fine with the TS lint disbale